tslint.json

```diff
@@ -1,6 +1,7 @@
 {
     "extends": ["@0x/tslint-config"],
     "rules": {
-        "custom-no-magic-numbers": [true, 0, 1, 2, 10]
+        "custom-no-magic-numbers": [true, 0, 1, 2, 10],
+        "no-unnecessary-type-assertion": false
```

Comment:
I had to abuse the non-null assertion operator (`!`) quite a bit in order to avoid having to do major refactors to fix our potentially undefined types sprinkled in several places. TSLint does not like this as it does not change the underlying type, just makes it non-nullable.

But at least we have a clear idea now where we are overriding potentially undefined properties as opposed to before where TS wasn't even checking that. In my opinion, this is strictly better, and we can make an effort to gradually clean this up.